aws/provider_test.go

```diff
@@ -1036,6 +1036,66 @@ func testAccErrorCheckSkipMessagesContaining(t *testing.T, messages ...string) r
 	}
 }
 
+type ServiceErrorCheckFunc func(*testing.T) resource.ErrorCheckFunc
+
+var serviceErrorCheckFuncs map[string]ServiceErrorCheckFunc
+
+func RegisterServiceErrorCheckFunc(endpointID string, f ServiceErrorCheckFunc) {
+	if serviceErrorCheckFuncs == nil {
+		serviceErrorCheckFuncs = make(map[string]ServiceErrorCheckFunc)
+	}
+	serviceErrorCheckFuncs[endpointID] = f
+}
+
+func testAccErrorCheck(t *testing.T, endpointIDs ...string) resource.ErrorCheckFunc {
+	return func(err error) error {
+		if err == nil {
+			return err
+		}
+
+		for _, endpointID := range endpointIDs {
+			if f, ok := serviceErrorCheckFuncs[endpointID]; ok {
+				ef := f(t)
+				err = ef(err)
+			}
+		}
+
+		if testAccErrorCheckCommon(err) {
+			t.Skipf("skipping test for %s/%s: %s", testAccGetPartition(), testAccGetRegion(), err.Error())
+		}
```

Comment:
In the case of looping through multiple services, error checking should occur at each service or break to save unnecessarily looping through additional services:

```suggestion
		for _, endpointID := range endpointIDs {
			if f, ok := serviceErrorCheckFuncs[endpointID]; ok {
				ef := f(t)
				err = ef(err)
			}

			if err == nil {
				break
			}
		}

		if testAccErrorCheckCommon(err) {
			t.Skipf("skipping test for %s/%s: %s", testAccGetPartition(), testAccGetRegion(), err.Error())
		}
```